shell/components/AppModal.vue

```diff
@@ -0,0 +1,109 @@
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+  name:  'AppModal',
+  props: {
+    /**
+     * If set to false, it will not be possible to close modal by clicking on
+     * the background or by pressing Esc key.
+     */
+    clickToClose: {
+      type:    Boolean,
+      default: true,
+    },
+    /**
+     * Width in pixels or percents (50, "50px", "50%").
+     *
+     * Supported string values are <number>% and <number>px
+     */
+    width: {
+      type:    [Number, String],
+      default: 600,
+    }
+  },
+  mounted() {
+    document.addEventListener('keydown', this.handleEscapeKey);
+  },
+  beforeDestroy() {
+    document.removeEventListener('keydown', this.handleEscapeKey);
+  },
+  methods: {
+    handleClickOutside(event: MouseEvent) {
+      if (
+        this.clickToClose &&
+        this.$refs.modalRef &&
+        !(this.$refs.modalRef as HTMLElement).contains(event.target as Node)
+      ) {
+        this.$emit('close');
+      }
+    },
+    handleEscapeKey(event: KeyboardEvent) {
+      if (this.clickToClose && event.key === 'Escape') {
+        this.$emit('close');
+      }
+    }
+  }
+});
+</script>
+
+<template>
+  <mounting-portal
```

Comment:
We use `mounting-portal` to allow us to define our portal target in a single location, `index.html`, that's not managed by Vue[^1]. 

While developing, I found that we would require multiple instances of `<portal-target name="modals">", one for each of our templates (`default.vue`, `home.vue`, etc...). I didn't like the repetition, so I opted to find a single location that would allow us to expose the modals target throughout the entire app. 

Additionally, there's some benefits to keeping our modal target closer to the end of `body`, namely a simplification of CSS[^2] 

[^1]: https://v2.portal-vue.linusb.org/guide/advanced.html#rendering-outside-of-the-vue-app
[^2]: https://v2.portal-vue.linusb.org/guide/getting-started.html#positioning-modals-overlays